widget/playground/src/utils/common.ts

```diff
@@ -0,0 +1,19 @@
+export function shallowEqual(
+  object1: { [x: string]: string },
+  object2: { [x: string]: string | undefined }
+) {
```

Comment:
You could rewrite signature like this to make it more generic:

```
export function shallowEqual<T>(
  object1: { [key: string]: T },
  object2: { [key: string]: T | undefined }
): boolean {
```

Also I didn't understand why `object1` values couldn't be undefined but `object2` could be.

Reply:
fixed